### PR TITLE
Fix currency format on order-pay page if currency was changed via switcher

### LINKED ISSUE #3318: Changing currency on the order-pay page changes currency format
### Describe the bug
If a customer is paying for an existing order, they are not able to change the currency of the order, but it's possible for the widget to display. If the currency is changed on the order pay page, the formatting of the order currency changes, even though the currency does not.

Note: Due to https://github.com/Automattic/woocommerce-payments/issues/3317 this can only be seen with the legacy currency switcher widget at the moment.

### To Reproduce

1. Add item to cart and proceed to checkout.
2. Use 4000000000000002 with any future expiration and cvv.
3. After order fails, go to My Account > Orders, and click to pay the order.
4. Change the currency via the (legacy) widget, and the format will change. It's easiest to see if the order is in USD and you switch to EUR.

### PR DESCRIPTION
**Pull request by jessepearson**
Fixes #3318

#### Changes proposed in this Pull Request

* Add action to `before_woocommerce_pay` so that `init_order_currency_from_query_vars` will be called in order to initialize the `order_currency` for us to use.
* Set `WC_Shortcode_Checkout::order_pay` as another backtrace call that should cause the currency to be overridden. This then allows us to use the `order_currency` property set.
* Update `create_payment_intent` to get the order currency instead of the selected currency if an order is passed. 

#### Testing instructions

1. Add item to cart and proceed to checkout using USD.
2. Use 4000000000000002 with any future expiration and cvv.
3. After order fails, go to My Account > Orders, and click to pay the order.
4. Change the currency via the (legacy) widget and change the currency to GBP or JPY.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

File: includes/multi-currency/FrontendCurrencies.php [hunk 1]
```diff
@@ -215,6 +216,16 @@ public function init_order_currency( $order ) {
...
 	}
 
+	/**
+	 * Gets the order id from the wp query_vars and then calls init_order_currency.
+	 *
+	 * @return void
+	 */
+	public function init_order_currency_from_query_vars() {
+		global $wp;
+		$this->init_order_currency( $wp->query_vars['order-pay'] );
```
Comment (tpaksu) [inline includes/multi-currency/FrontendCurrencies.php:226]: I would add a check if the `query_vars` has an `order-pay` key first here before sending it to the function. What do you think?

**Review by tpaksu — APPROVED** [review]
Tested and approved, the currency of the checkout form doesn't change when you change it from any currency switcher. LGTM, you can :ship: it.

File: includes/multi-currency/FrontendCurrencies.php (hunk 1 above)
Comment (jessepearson) [reply includes/multi-currency/FrontendCurrencies.php]: I think it's a good idea, and updated in https://github.com/Automattic/woocommerce-payments/pull/3323/commits/1d703be4bbd253c28b5ad09735410b4f457a2363 . 

The reason I didn't do it initially is due to WooCommerce core _shouldn't_ call the `before_woocommerce_pay` action if the `order-pay` key is empty: https://github.com/woocommerce/woocommerce/blob/5.9.0/includes/shortcodes/class-wc-shortcode-checkout.php#L57-L80 . It's better to double check, though, thanks for spotting this.

Comment (tpaksu) [reply includes/multi-currency/FrontendCurrencies.php]: Thanks for doing this :) IMO, It's good to be extra defensive in those cases.

**Pull request merged**